scripts/exobsforge_global_atmos_bufr_dump.py

```diff
@@ -12,6 +12,60 @@
 logger = Logger(level='DEBUG', colored_log=True)
 
 
+def merge_observation_defaults(task_yaml, section_name, default_obs=None):
+    """
+    Merge default observation configs into a YAML-loaded task dict.
+    Parameters
+    ----------
+    task_yaml : dict
+        The full YAML-loaded dictionary.
+    section_name : str
+        The top-level key containing observations (e.g., 'atmosbufrdump').
+    default_obs : dict, optional
+        Default observation parameters. If None, uses a standard default.
+    Returns
+    -------
+    dict
+        Updated YAML dict with merged defaults.
+    """
+    if default_obs is None:
+        default_obs = {
+            "method": "script2netcdf",
+            "input_file": None,
+            "output_file": None,
+            "mapping_file": None,
+            "script_file": None,
+        }
+
+    section = task_yaml.get(section_name, {})
+    observations = section.get('observations', {})
+    logger.debug(f"Merging defaults into section '{section_name}' with observations: {list(observations.keys())}")
+
+    merged_observations = {}
+    for obs_name, obs_cfg in observations.items():
+        logger.debug(f"Merging observation '{obs_name}' with config: {obs_cfg}")
+        obs_cfg = obs_cfg or {}  # handle None or empty dict
+        merged_cfg = {**default_obs, **obs_cfg}
+
+        # Fill dynamic defaults based on obs_name
+        if merged_cfg["input_file"] is None:
+            merged_cfg["input_file"] = f"{obs_name}.tm00.bufr_d"
+        if merged_cfg["output_file"] is None:
+            merged_cfg["output_file"] = f"{obs_name}.tm00.nc"
```

Comment:
We should make the defaults not include the `.tm00` as that is a deprecated convention

Reply:
both of these, yes